### DISCUSSION TITLE
add a "repeat" mod

### DISCUSSION THREAD

**Discussion by e-23a** [ppy/osu#27355] [Ideas]
they should like add a "repeat" mod where you can choose a value and then after the beatmap ends it automatically loops so you can play it again, another value can be named like "rest" and it serves after the beatmap ends (it can be skipped), in how many seconds you want to rest until it loops. and finally exiting the loop can be easily done by just escaping as it will stop the mod.

**Comment by trustytrojan** [comment]
I like the idea but that doesn't really modify gameplay so it isn't really a "mod" per se. Holding backtick (or whatever your retry key is) when you finish a map does exactly what you're looking for.

**Reply by OwenCMYK** [reply]
A lot of the fun mods don't modify gameplay at all. That doesn't seem like it's really a requirement for a mod to be included in the game

**Reply by trustytrojan** [reply]
By "modify gameplay" I mean changing the sensory input that the player receives and seeing if they can produce the same results. The fun mods do modify gameplay in that sense because you have to take in a different visual picture and adapt.

The reason I say the "repeat mod" isn't a mod is because during gameplay the visual input you receive is the same throughout the entire map. All that this "repeat mod" does is automatically hit the retry key for you. It should really just be a keybind toggle instead of a mod.

**Reply by OwenCMYK** [reply]
I think the idea is more that it will add extra objects to the end of the beatmap and loop the song. So it's not just that it's restarting the beatmap, but rather than the beatmap doesn't end until you've done everything however many times you have set. You would then presumably be scored based on the multiple loops as one.
